kubectl-plugin/pkg/cmd/job/job_submit.go

```diff
@@ -421,55 +429,60 @@ func (options *SubmitJobOptions) Run(ctx context.Context, factory cmdutil.Factor
-	defer cancel()
-	go func() {
-		fmt.Printf("Port Forwarding service %s\n", svcName)
-		if err := portForwardCmd.ExecuteContext(portforwardctx); err != nil {
-			log.Fatalf("Error occurred while port-forwarding Ray dashboard: %v", err)
+	if options.address == "" {
+		svcName, err := k8sClients.GetRayHeadSvcName(ctx, options.namespace, util.RayCluster, options.cluster)
+		if err != nil {
+			return fmt.Errorf("Failed to find service name: %w", err)
 		}
-	}()
 
-	// Wait for port forward to be ready
-	var portforwardReady bool
-	portforwardWaitStartTime := time.Now()
-	currTime = portforwardWaitStartTime
+		// start port forward section
+		portForwardCmd := portforward.NewCmdPortForward(factory, *options.ioStreams)
+		portForwardCmd.SetArgs([]string{"service/" + svcName, fmt.Sprintf("%d:%d", 8265, 8265)})
+
+		// create new context for port-forwarding so we can cancel the context to stop the port forwarding only
+		portForwardCtx, cancel := context.WithCancel(ctx)
+		defer cancel()
+		go func() {
+			fmt.Printf("Port forwarding service %s\n", svcName)
+			if err := portForwardCmd.ExecuteContext(portForwardCtx); err != nil {
+				log.Fatalf("Error occurred while port-forwarding Ray dashboard: %v", err)
+			}
+		}()
 
-	portforwardCheckRequest, err := http.NewRequestWithContext(ctx, http.MethodGet, dashboardAddr, nil)
-	if err != nil {
-		return fmt.Errorf("Error occurred when trying to create request to probe cluster endpoint: %w", err)
-	}
-	httpClient := http.Client{
-		Timeout: 5 * time.Second,
-	}
-	fmt.Printf("Waiting for portforwarding...")
-	for !portforwardReady && currTime.Sub(portforwardWaitStartTime).Seconds() <= portforwardtimeout {
-		time.Sleep(2 * time.Second)
-		rayDashboardResponse, err := httpClient.Do(portforwardCheckRequest)
+		// Wait for port forward to be ready
+		var portForwardReady bool
+		portForwardWaitStartTime := time.Now()
+		currTime = portForwardWaitStartTime
+
+		portforwardCheckRequest, err := http.NewRequestWithContext(ctx, http.MethodGet, dashboardAddr, nil)
 		if err != nil {
-			err = fmt.Errorf("Error occurred when waiting for portforwarding: %w", err)
-			fmt.Println(err)
+			return fmt.Errorf("Error occurred when trying to create request to probe cluster endpoint: %w", err)
 		}
-		if rayDashboardResponse.StatusCode >= 200 && rayDashboardResponse.StatusCode < 300 {
-			portforwardReady = true
+		httpClient := http.Client{
+			Timeout: 5 * time.Second,
 		}
-		rayDashboardResponse.Body.Close()
-		currTime = time.Now()
-	}
-	if !portforwardReady {
-		return fmt.Errorf("Timed out waiting for port forwarding")
+		fmt.Printf("Waiting for port forwarding...")
+		for !portForwardReady && currTime.Sub(portForwardWaitStartTime).Seconds() <= portForwardTimeout {
+			time.Sleep(2 * time.Second)
+			rayDashboardResponse, err := httpClient.Do(portforwardCheckRequest)
+			if err != nil {
+				err = fmt.Errorf("Error occurred when waiting for port forwarding: %w", err)
+				fmt.Println(err)
+			}
```

Comment:
```suggestion
			if err != nil {
				err = fmt.Errorf("Error occurred when waiting for port forwarding: %w", err)
				fmt.Println(err)
				currTime = time.Now()
				continue
			}
```

Prevent panic when `rayDashboardResponse` is nil.